backend/business-partner-agent/src/main/java/org/hyperledger/bpa/controller/IssuerController.java

```diff
@@ -0,0 +1,80 @@
+package org.hyperledger.bpa.controller;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.*;
+import io.micronaut.scheduling.TaskExecutors;
+import io.micronaut.scheduling.annotation.ExecuteOn;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import io.micronaut.validation.Validated;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.hyperledger.bpa.api.aries.SchemaAPI;
+import org.hyperledger.bpa.controller.api.issuer.CreateCredDefRequest;
+import org.hyperledger.bpa.controller.api.issuer.CreateSchemaRequest;
+import org.hyperledger.bpa.impl.IssuerManager;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Controller("/api/issuer")
+@Tag(name = "Credential Issuance Management")
+@Validated
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@ExecuteOn(TaskExecutors.IO)
+public class IssuerController {
+
```

Comment:
Yes, we do have 2 concerns right now: add/import schema from Ledger and the creation of a new schema by the BPA. When we have overlapping functionality (read, list, delete, etc), we can move the code into the service.  For the first iteration, I am just putting together all the functionality an Issuer needs, I'm not 100% sure that the Schema API objects will be the same between Admin and Issuer, but we will see and refactor, yes, exact logic needs to be done in one place.  
At the controller level, I do think we can have the same functions under various endpoints. It will simplify security and it makes it easy to identify all the functions one needs for their task, instead of knowing that something is under /admin and something is under /issuer etc. 
We definitely need to refactor once we have more functionality and ensure we have business logic coded once and re-used where appropriate.

Reply:
Makes sense. My thinking is that from the perspective of the business logic both operations are tight together and what we have is basically /addExistingSchema and /createNewSchema. So I would propose to keep all operations related to schema management in the schema manager. For the API we can keep things separated if we think they indeed are, or another option would be to create a schema controller. The only thing that is a bit dangerous right now is that one api also affects the other, and there are no tests to see if there are side effects.

Reply:
Ok, I see your point; makes sense to me. Yes, I will move the logic for create schema into schema manager.

Reply:
Code moved to SchemaService